packages/react-components/react-rating-preview/src/components/RatingItem/useRatingItem.tsx

```diff
@@ -14,15 +14,15 @@ import { useRatingContextValue_unstable } from '../../contexts/RatingContext';
  * @param ref - reference to root HTMLElement of RatingItem
  */
 export const useRatingItem_unstable = (props: RatingItemProps, ref: React.Ref<HTMLSpanElement>): RatingItemState => {
-  const context = useRatingContextValue_unstable();
+  const context = useRatingItemContextValue_unstable();
   const { value = 0 } = props;
 
-  const ratingValue = context?.value || 0;
+  const ratingValue = context.value || 0;
```

Comment:
Since RatingDisplay allows values that aren't a multiple of 0.5, this needs to be rounded to the nearest 0.5.
```suggestion
  const ratingValue = Math.round((context.value || 0) * 2) / 2; // round to the nearest 0.5
```